chore/unit-tests-for-assertion-module

kylebeee opened the pull request:
Description
- added unit tests for assertion module controller
- removed express response object where possible to avoid mocking
- Jira: [ENG-280](https://algorandfoundation.atlassian.net/browse/ENG-280)

[ENG-280]: https://algorandfoundation.atlassian.net/browse/ENG-280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

kylebeee committed:
chore/unit-tests-for-assertion-module

- added unit tests for assertion module controller
- removed express response object where possible to avoid mocking

PhearZero reviewed (changes requested):
- remove unnecessary `await` on sync service calls
- remove `stringify` of return statements
~~- add missing AssertionService unit tests~~

kylebeee committed:
- removed unnecessary await (my mistake, got confused when i mocked a
  resolved value for it)
- updated endpoints to not JSON.stringify responses ( handled by nest )
- removed unnecessary mocks

kylebeee committed:
- removed extra dummyOptions declaration & removed assertionService mock
  overrides

kylebeee committed:
removed JSON.stringify from expect assertion controller results

kylebeee committed:
final adjustments, removed await & spread mocks into useValue

kylebeee committed:
change toThrowError to ToThrow(HttpException)

kylebeee committed:
- changed mock args & responses to more verbose naming
- changed mock challenges to be more realistic

kylebeee committed:
broke out mock overrides to hooks instead of inside tests

kylebeee committed:
modified organization of unit tests to be clearer about mocks

kylebeee committed:
changed hardcoded buffer to randomBytes

kylebeee committed:
switch from generic http exceptions to with status codes

kylebeee committed:
fixed imports for a few exceptions
